modules/common/networking/hosts.nix

```diff
@@ -0,0 +1,74 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  ...
+}: let
+  # please note that .100. network is not
+  # reachable from ghaf-host. It's only reachable
+  # guest-to-guest.
+  # Use to .101. (debug) to access guests from host.
+  # debug network hosts are post-fixed: <hostname>-debug
+  ipBase = "192.168.100";
+  debugBase = "192.168.101";
+  hostsEntries = [
+    {
+      ip = 1;
```

Comment:
One generic suggestion, instead of using hardcoded numbers here and in other files [loc1](https://github.com/tiiuae/ghaf/pull/584/files#diff-477e661df628667d7a61761a39c61bfb59fa57fdba1e3b7e3e58468d2f5b396dR24), [loc2](https://github.com/tiiuae/ghaf/pull/584/files#diff-ed0b8ba6a36a522c74a303a9dc0f1c41cfc3dc6a07ef06450aabf6df1318db8aR32)   maybe we can associate those values with nix options, example instead of ip = 1 we can use ip = `config.ghaf.ip.index.netvm`  so it will be easy to access/read across different files and easy to maintain :)
Not necessary to be in scope of this PR but can be taken up later.

Reply:
Original idea was to try to use the list of appvms to generate hosts file, but hosts file gets generated while the vms are created, so there was a bit of an chicken and egg problem.
Right away, even with the help of some nix experts, we could not figure out a reasonably dynamic way to handle this, so at least for now we decided to have some redundancy (several lists of vms) so we get this thing working. A more elegant solution can be implemented later.

Reply:
I think this is good for now.
Maybe consider updating the ipBase to a separate config variable. The issue is that there is are a few "magic" values with the ipbase / gateway address in various .nix files (at least in hosts.nix and vm-networking.nix).
I agree with the difficulty of handling this dynamically (that is what dhcp is for). In our use Dhcp just causes its own issues that would be really complex to solve.

Reply:
What I was referring to by "dynamic" was "build-time-dynamic" solution, so that we'd have one list of vms in one place, and both the actual vms and hosts entries would get generated from that. Not referring to dynamic host configuration protocol here.

Reply:
Yeah, we discussed to get the scattered/distributed networking configuration refactoring in the next phase. The coming centralised networking configuration nix module could look like [target hw definitions](https://github.com/tiiuae/ghaf/blob/main/modules/hardware/lenovo-x1/definitions/x1-gen11.nix) but with MACs, IPs, network bases etc. - collected to a single file and referenced elsewhere via config. Anyone configuring the ghaf target network would see all the values in one file. That structural change is less prone to regression than this and can bring some build-time checks via value references.

This PR mainly addresses the transition from internal DNS/DHCP with dnsmasq to static internal network with names.